File: tournaments/isTandemRepeat/isTandemRepeat.cpp
bool isTandemRepeat(std::string inputString) {

  int len = inputString.size();
  if (len % 2 == 1) {
    return false;
  }
  for (int i = 0; i < len / 2; i++) {
    if (inputString[i] != inputString[i + len / 2]) {
      return false;
    }
  }

  return true;
}

File: tournaments/isTandemRepeat/isTandemRepeat.js
function isTandemRepeat(inputString) {
    if (inputString.length & 1) {
        return false;
    }

    for (let i = 0; (inputString.length >> 1) > i; ++i) {
        if (inputString[i] !== inputString[(inputString.length >> 1) + i]) {
            return false;
        }
    }

    return true;
}
